### ISSUE TITLE
[bug] False negatives for `no-missing-key` since 2.6.4 version

### ISSUE THREAD

**Issue by pawelblaszczyk5** [Rel1cx/eslint-react#1412]
### Describe the bug

Hello,
I've updated version of my plugin and spotted some false negatives for `no-missing-key`. I pinpointed it to break between 2.6.2 and 2.6.4 (2.6.3 wasn't released if I saw correctly, I also verified the newest beta but it doesn't change anything ere). The exact issue for me is using pattern matching utility from `Effect`, which gets flagged:

```tsx
 {Match.value(y).pipe(
        Match.when(true, () => <div>Test</div>),
        Match.when(false, () => <div>Test</div>),
        Match.exhaustive
      )}
```

<img width="1186" height="491" alt="Image" src="https://github.com/user-attachments/assets/b16630a9-5616-4513-ac85-480c6b5955bf" />

in that case - both of this `div` gets flagged with error. I've looked through commits and it seems like this one was released exactly in this version https://github.com/Rel1cx/eslint-react/commit/8361a356c4d2ed98153fc0130cc6ae12aa08a7d2

### Reproduction

https://github.com/pawelblaszczyk5/react-eslint-x-array-misdetection-repro

### Expected behavior

No array mapping isn't flagged by the rule

### Platform and versions

```Text
As above + 25.2.0 node
```

### Stack trace

```Text

```

**Comment by Rel1cx** [comment]
> I've looked through commits and it seems like this one was released exactly in this version 8361a35

You are right! Thank you for your detailed report. It's very helpful.

**Comment by pawelblaszczyk5** [comment]
> > I've looked through commits and it seems like this one was released exactly in this version [8361a35](https://github.com/Rel1cx/eslint-react/commit/8361a356c4d2ed98153fc0130cc6ae12aa08a7d2)
> 
> You are right! Thank you for your detailed report. It's very helpful.

My pleasure, thank you very much for fixing and overall for the library - awesome work! 

Btw - did you thought about setting up sponsorships? I’d be happy to thank you for the job you’re doing here this way.
